src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -99,7 +95,7 @@ public CompilationResult Compile(RelativeFileInfo fileInfo, string compilationCo
 
             var startTimestamp = _logger.IsEnabled(LogLevel.Debug) ? Stopwatch.GetTimestamp() : 0;
 
-            var assemblyName = Path.GetRandomFileName();
+            var assemblyName = fileInfo.FileInfo.Name + "." + Path.GetRandomFileName();
```

Comment:
It made debugging in the Modules window a tad simpler. I'll get this tracked in a separate work item